### ISSUE TITLE
prompt_settings.yaml cant be found??

### ISSUE THREAD

**Issue by FreddzieAI** [Significant-Gravitas/Auto-GPT#4598]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### Which Operating System are you using?

Docker

### Which version of Auto-GPT are you using?

Latest Release

### Do you use OpenAI GPT-3 or GPT-4?

GPT-4

### Which area covers your issue best?

Installation and setup

### Describe your issue.

![image](https://github.com/Significant-Gravitas/Auto-GPT/assets/132779218/022e68cd-0c57-474c-9293-4634b5b508a3)

this error pops up even tho i have the file

**Comment by FreddzieAI** [comment]
i can use this command as a work arround but is there a way to not have to do this
docker-compose run --rm -v /home/Freddie_Test-7/Auto-GPT-0.4.0/prompt_settings.yaml:/app/prompt_settings.yaml auto-gpt

**Comment by adoultremont** [comment]
Had this issue and just copy/pasted into ./prompt_settings.yaml

`constraints: [
  '~4000 word limit for short term memory. Your short term memory is short, so immediately save important information to files.',
  'If you are unsure how you previously did something or want to recall past events, thinking about similar events will help you remember.',
  'No user assistance',
  'Exclusively use the commands listed below e.g. command_name'
]
resources: [
  'Internet access for searches and information gathering.',
  'Long Term memory management.',
  'GPT-3.5 powered Agents for delegation of simple tasks.',
  'File output.'
]
performance_evaluations: [
  'Continuously review and analyze your actions to ensure you are performing to the best of your abilities.',
  'Constructively self-criticize your big-picture behavior constantly.',
  'Reflect on past decisions and strategies to refine your approach.',
  'Every command has a cost, so be smart and efficient. Aim to complete tasks in the least number of steps.',
  'Write all code to a file.'
]`

**Comment by TsundokuJim** [comment]
Same issue. The prompt_settings.yaml file is sitting in the main Auto-GPT-0.4.0 directory and contains the usual constraints, resources, and performance eval settings, yet auto-gpt can't find it. It's correctly identified by Windows as a yaml source file and isn't tagged as read-only.

**Comment by maxfield-allison** [comment]
same issue on linux in docker.

**Comment by ericflecher** [comment]
Same issue

**Comment by Naetmul** [comment]
I'm currently running WSL2 Docker on a Windows system.

I took the `prompt_settings.yaml` file from the repository and moved it to the `Auto-GPT` folder, the same place where the `.env` file is located. After that, I added the line `- ./prompt_settings.yaml:/app/prompt_settings.yaml` under the `volumes: section` in the `docker-compose.yml` file.

```yaml
version: "3.9"
services:
  auto-gpt:
    image: significantgravitas/auto-gpt
    env_file:
      - .env
    profiles: ["exclude-from-up"]
    volumes:
      - ./auto_gpt_workspace:/app/autogpt/auto_gpt_workspace
      - ./data:/app/data
      - ./logs:/app/logs
      - ./prompt_settings.yaml:/app/prompt_settings.yaml
```

Following these modifications, I ran the command `docker-compose build auto-gpt`. It works after these steps.

**Comment by playfluent** [comment]
this was able to fix my issue running Docker on MacOS

**Comment by hugh-min** [comment]
same issue on Windows Docker

**Comment by ccdanpian** [comment]
I had the same problem when using docker pull. 
You can try to git clone to the local, just modify the .env, then execute docker-compose build and run docker-compose run

**Comment by ccdanpian** [comment]
I think the problem may be that the docker image and the code version downloaded from git are not consistent. The easiest way is not to use docker pull to pull the image from the server, git clone it directly to local, don’t need to modify the docker-compose.yml.
this method works on both of my computers

```
step1: git clone -b stable https://github.com/Significant-Gravitas/Auto-GPT.git
don't modify docker-compose.yml, just goto step2
setp2: cd Auto-GPT
step3: modify the .env
step4: docker-compose build auto-gpt
step5: docker-compose run --rm auto-gpt
```

**Comment by ZaneStanutz** [comment]
docker image seems to be broken for me as well .. github repo worked fine. thx @ccdanpian !

**Comment by cap273** [comment]
Docker image was also broken for me. Thank you @ccdanpian !

**Comment by lmsanch** [comment]
I ran auto locally on my M2 mac, and also using a Docker container using the solution of @ccdanpian above and using the stable release here https://github.com/Significant-Gravitas/Auto-GPT/releases/tag/v0.4.0
In both cases, I get this:
```
Using memory of type:  JSONFileMemory
Using Browser:  chrome
FAILED FILE VALIDATION  The file `prompt_settings.yaml` wasn't found
DOUBLE CHECK CONFIGURATION  Please ensure you've setup and configured everything correctly. Read https://github.com/Torantulino/Auto-GPT#readme to double check. You can also create a github issue or join the discord and ask there!
```
Any help?

**Comment by ccdanpian** [comment]
@lmsanch 
Sorry, I don’t have a Mac, so I haven’t tried it on a Mac. The following configuration also works well ok on my win11 computer.
Please try edit your docker-compose.yml follow the attached file, and try again. 
[docker-compose.zip](https://github.com/Significant-Gravitas/Auto-GPT/files/11740425/docker-compose.zip)

**Comment by lmsanch** [comment]
I used that file and replicated  the whole thing. This is what I get:
```
Using memory of type:  JSONFileMemory
Using Browser:  chrome
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/app/autogpt/__main__.py", line 5, in <module>
    autogpt.cli.main()
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1635, in invoke
    rv = super().invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/app/autogpt/cli.py", line 96, in main
    run_auto_gpt(
  File "/app/autogpt/main.py", line 183, in run_auto_gpt
    system_prompt = ai_config.construct_full_prompt()
  File "/app/autogpt/config/ai_config.py", line 141, in construct_full_prompt
    prompt_generator = build_default_prompt_generator()
  File "/app/autogpt/prompts/prompt.py", line 32, in build_default_prompt_generator
    prompt_config = PromptConfig(CFG.prompt_settings_file)
  File "/app/autogpt/config/prompt_config.py", line 42, in __init__
    (validated, message) = utils.validate_yaml_file(config_file)
  File "/app/autogpt/utils.py", line 65, in validate_yaml_file
    with open(file, encoding="utf-8") as fp:
IsADirectoryError: [Errno 21] Is a directory: 'prompt_settings.yaml'
```

**Comment by ccdanpian** [comment]
@lmsanch 
I see that there is already an [FR#4680](https://github.com/Significant-Gravitas/Auto-GPT/pull/4680) to fix this bug, so it should be working normally soon.

**Comment by dsdnklasmals** [comment]
> docker image seems to be broken for me as well .. github repo worked fine. thx @ccdanpian !
> 
> Docker的形象对我来说似乎也被打破了。GitHub Repo运行良好。thx @ccdanpian！

docker 是我这辈子用过最垃圾的东西 没有之一！！！

**Comment by abautista-caylent** [comment]
> i can use this command as a work arround but is there a way to not have to do this
> docker-compose run --rm -v /home/Freddie_Test-7/Auto-GPT-0.4.0/prompt_settings.yaml:/app/prompt_settings.yaml auto-gpt

thank you!

**Comment by ntindle** [comment]
Fixed by #4680

**Comment by Luckiluciano** [comment]
> [ccdanpian](/ccdanpian)

I am not a developer.  I am simply an enthusiast.  I compiled all the possible solutions including, the one's outside of this thread and fed it to ChatGPT.  ChatGPT stated that this one is the best solution.  After a clean wipe, I followed these instructions and had a succesful install.
